File: problems/1124_longest_well_performing_interval/solution.py
class Solution:
    def longestWPI(self, hours: list[int]) -> int:
        score = 0
        result = 0
        first = {}
        for i, h in enumerate(hours):
            score += 1 if h > 8 else -1
            if score > 0:
                result = i + 1
            else:
                if score not in first:
                    first[score] = i
                if score - 1 in first:
                    result = max(result, i - first[score - 1])
        return result

File: problems/1124_longest_well_performing_interval/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int longestWPI(vector<int>& hours) {
        int score = 0, result = 0;
        unordered_map<int, int> first;
        for (int i = 0; i < hours.size(); i++) {
            score += hours[i] > 8 ? 1 : -1;
            if (score > 0) {
                result = i + 1;
            } else {
                if (!first.count(score)) first[score] = i;
                if (first.count(score - 1)) {
                    result = max(result, i - first[score - 1]);
                }
            }
        }
        return result;
    }
};
